### ISSUE TITLE
[Bug]: Vue 3.4 - props and events are not inferred from `defineModel`

### ISSUE THREAD

**Issue by Vercadium** [storybookjs/storybook#25626]
### Describe the bug

Since `defineModel` became stable in Vue 3.4, [this is now the recommended approach](https://vuejs.org/guide/components/v-model.html) for using `v-model`.

Ideally Storybook would understand that `defineModel` is constructing the props/events for these (example below) and display them in the `argTypes` automatically. It would be preferred if both the type and JSDoc comment could be understood also.

_After 3.4:_
```vue
<script setup>
const model = defineModel();
</script>

<template>
  <input v-model="model" />
</template>
```

_Before 3.4:_
```vue
<script setup>
const props = defineProps(['modelValue']);
const emit = defineEmits(['update:modelValue']);
</script>

<template>
  <input
    :value="props.modelValue"
    @input="emit('update:modelValue', $event.target.value)"
  />
</template>
```

**Comment by shilman** [comment]
@chakAs3 @kasperpeulen will this be fixed moving the argtype inference to volar? https://github.com/storybookjs/storybook/pull/22285

**Comment by larsrickert** [comment]
@shilman Yes and no :D I quickly tested it out, with Volar the property will be added but the JSDoc / description and the corresponding event is missing. However, this feature is currently missing in Volar itself.

I added a feature request for Volar about this: https://github.com/vuejs/language-tools/issues/3894

**Comment by vanessayuenn** [comment]
Looks like the feature request has been completed and released by Volar! Thanks all. Closing this issue.
